### PR TITLE
[Merged by Bors] - style: cleanup some autoImplicits

### PR DESCRIPTION
**Pull request by eric-wieser**
In some cases adding the arguments manually results in a more obvious argument order anyway

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/Defs.lean [hunk 1]
```diff
@@ -93,13 +93,11 @@ attribute [to_additive existing (reorder := 1 2, 5 6) hSMul] HPow.hPow
 attribute [to_additive existing (reorder := 1 2, 4 5) smul] Pow.pow
 
 @[to_additive (attr := default_instance)]
-instance instHSMul [SMul α β] : HSMul α β β where
+instance instHSMul {α β} [SMul α β] : HSMul α β β where
   hSMul := SMul.smul
 
 attribute [to_additive existing (reorder := 1 2)] instHPow
```
Comment (eric-wieser) [inline Mathlib/Algebra/Group/Defs.lean:99]: In this example, the `reorder` is easier to read when you can actually see the arguments it refers to

File: Mathlib/Algebra/Invertible/Defs.lean [hunk 2]
```diff
@@ -241,26 +238,27 @@ def Invertible.mul [Monoid α] {a b : α} (_ : Invertible a) (_ : Invertible b)
...
   rw [← mul_left_inj_of_invertible (c := ⅟c), invOf_mul_self_assoc]
 
-theorem mul_invOf_eq_iff_eq_mul_right [Monoid α] [Invertible (c : α)] :
+
+theorem mul_invOf_eq_iff_eq_mul_right [Monoid α] {a b c : α} [Invertible (c : α)] :
     a * ⅟c = b ↔ a = b * c := by
   rw [← mul_right_inj_of_invertible (c := c), mul_invOf_mul_self_cancel]
 
-theorem mul_right_eq_iff_eq_mul_invOf [Monoid α] [Invertible (c : α)] :
+theorem mul_right_eq_iff_eq_mul_invOf [Monoid α] {a b c : α} [Invertible (c : α)] :
```
Comment (eric-wieser) [inline Mathlib/Algebra/Invertible/Defs.lean:262]: Without these the argument order was `c a b`, which is arguably weird.

**Review by fpvandoorn — APPROVED** [review]

File: Mathlib/Algebra/Invertible/Defs.lean [hunk 3]
```diff
@@ -241,26 +238,27 @@ def Invertible.mul [Monoid α] {a b : α} (_ : Invertible a) (_ : Invertible b)
   invertibleMul _ _
 #align invertible.mul Invertible.mul
 
-theorem mul_right_inj_of_invertible [Monoid α] (c : α) [Invertible c] :
+theorem mul_right_inj_of_invertible [Monoid α] {a b : α} (c : α) [Invertible c] :
     a * c = b * c ↔ a = b :=
   ⟨fun h => by simpa using congr_arg (· * ⅟c) h, congr_arg (· * _)⟩
 
-theorem mul_left_inj_of_invertible [Monoid α] (c : α) [Invertible c] :
+theorem mul_left_inj_of_invertible [Monoid α] {a b : α} (c : α) [Invertible c] :
```
Comment (YaelDillies) [inline Mathlib/Algebra/Invertible/Defs.lean:245]: Can you put all these lemmas in a section with `variable [Monoid α] {a b : α} (c : α) [Invertible c]`?

**Comment by fpvandoorn** [comment]
bors d+

File: Mathlib/Algebra/Invertible/Defs.lean (hunk 3 above)
Comment (eric-wieser) [reply Mathlib/Algebra/Invertible/Defs.lean:245]: This makes the diff much bigger; and I don't want anyone to be saying "look how much mess it makes to clear up autoimplicit.

I'll make a follow-up PR.

bors merge

**Pull request closed without merging**